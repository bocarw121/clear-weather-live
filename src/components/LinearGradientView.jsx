import React from "react";
import { LinearGradient } from "expo-linear-gradient";
import { StyleSheet, View, ScrollView } from "react-native";

import colors from "../constants/colors";

const LinearGradientView = ({ children }) => {
  return (
    <View>
      <LinearGradient
        colors={colors.linearGradient}
        style={[styles.linearGradient]}
        start={{ x: 0, y: 0 }}
        end={{ x: 1, y: 1 }}
      >
        <ScrollView>{children}</ScrollView>
      </LinearGradient>
    </View>
  );
};

export default LinearGradientView;

const styles = StyleSheet.create({
  linearGradient: {
    height: "100%",
  },
});
