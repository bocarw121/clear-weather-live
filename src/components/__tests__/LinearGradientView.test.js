import React from "react";

import { render } from "@testing-library/react-native";

import LinearGradientView from "../LinearGradientView";

it("Should render the LinearGradientView", () => {
  const screen = render(<LinearGradientView />);

  expect(screen).toMatchSnapshot();
});
