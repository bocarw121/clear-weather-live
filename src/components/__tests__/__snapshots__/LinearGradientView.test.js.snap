// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the LinearGradientView 1`] = `
<View>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278224127,
          4279857894,
        ],
        "endPoint": Array [
          1,
          1,
        ],
        "locations": undefined,
        "startPoint": Array [
          0,
          0,
        ],
      }
    }
    style={
      Array [
        Object {
          "height": "100%",
        },
      ]
    }
  >
    <RCTScrollView>
      <View />
    </RCTScrollView>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
